grpc-api/src/main/java/io/deephaven/grpc_api/session/SessionState.java

```diff
@@ -277,7 +328,7 @@ public void onExpired() {
      */
     @Override
     protected void destroy() {
-        expiration = null;
+        onExpired();
```

Comment:
Note that onExpired is idempotent -- but its purpose is to clear all outstanding state and to prevent additional outstanding state from being created. It doesn't matter if this session "disappears" because it times out (client fails to rotate session cookie) or via the closeSession rpc. In both cases we should prevent additional state from existing and we should close all outstanding streams at the moment of closure/expiration.

So I think this is not just OK, but adds value. Alternatively, we could have onExpired call destroy() and have the clean up logic inside of destroy(); maybe this is more the pattern you would want to see?

Reply:
I think having `destroy` (that is, being a `LivenessArtifact` muddies the water a bit. Best to just let `onExpired` do the work.